### ISSUE TITLE
Postgres BulkUpdate issues

### ISSUE THREAD

**Issue by SebastianBienert** [borisdj/EFCore.BulkExtensions#717]
Hi, I'm having troubles while running BulkUpdate concurrently on many service instances. When I run just one instance of application everything works fine, problems occur only when there are many instances accessing the same table. The exception thrown says:

```
 Exception data:
    Severity: ERROR
    SqlState: 55000
    MessageText: index "tempUniqueIndex_item_id" is not valid
    Position: 34
    File: parse_utilcmd.c
    Line: 2108
    Routine: transformIndexConstraint
```

I analyzed the code of PostrgresAdapter and I can't understand why do we need to create temporary unique index constraint on the updated table ([code](https://github.com/borisdj/EFCore.BulkExtensions/blob/4d4bf55cd3832fa7411605ee343aa7dce25c5ba7/EFCore.BulkExtensions/SQLAdapters/PostgreSql/PostgreSqlAdapter.cs#L179)). Also code for checking for unique constraints is a bit confusing ([code](https://github.com/borisdj/EFCore.BulkExtensions/blob/4d4bf55cd3832fa7411605ee343aa7dce25c5ba7/EFCore.BulkExtensions/SQLAdapters/PostgreSql/SqlQueryBuilderPostgreSql.cs#L130)). On the first part of the query It chceks only 'UNIQUE' constraint_types, so the primary keys are not included, then it appends AND clause with pirmary key column names, but primary key won't have 'UNIQUE' constraint type. I'm not sure what was the intention behind this query so I don't know if it has flaws.

**Comment by borisdj** [comment]
Postgres for _BulkUpdate_ uses **ON CONFLICT** which requires _UniqueIndex_ on _UpdateBy_ cols.
Otherwise throws: 'there is no unique or exclusion constraint matching the ON CONFLICT specification'.
If it already exist then no need to create it, and if UpdateBy column is PK then it will work even without UniqueIndex since PK has unique constrain by default.

**Comment by SebastianBienert** [comment]
From my understanding the piece of code checking for unique constraints does not take into account PK:
([code](https://github.com/borisdj/EFCore.BulkExtensions/blob/4d4bf55cd3832fa7411605ee343aa7dce25c5ba7/EFCore.BulkExtensions/SQLAdapters/PostgreSql/SqlQueryBuilderPostgreSql.cs#L130))
For example given entites from [InsertTestPostgreSql ](https://github.com/borisdj/EFCore.BulkExtensions/blob/4d4bf55cd3832fa7411605ee343aa7dce25c5ba7/EFCore.BulkExtensions.Tests/EFCoreBulkTest.cs#L26) test we would have query that looks like this:

```
SELECT COUNT(*) FROM INFORMATION_SCHEMA.TABLE_CONSTRAINTS tc 
INNER JOIN INFORMATION_SCHEMA.CONSTRAINT_COLUMN_USAGE cu ON cu.CONSTRAINT_NAME = tc.CONSTRAINT_NAME 
WHERE tc.CONSTRAINT_TYPE = 'UNIQUE' AND tc.TABLE_NAME = 'Item' AND cu.COLUMN_NAME = 'ItemId' 
```
For which response is 0, in the end resulting in creation of the unique index on ItemId. Primary keys constraints have 'PRIMARY KEY' CONSTRAINT_TYPE column so they are not accounted in this query even though PK is enough for the ON CONFLICT specification

**Comment by borisdj** [comment]
Fixed with v6.3.0
